rcp/org.testeditor.rcp4.views.tcltestrun.tests/src/org/testeditor/rcp4/views/tcltestrun/model/TestLogGroupBuilderTest.xtend

```diff
@@ -0,0 +1,207 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.tcltestrun.model
+
+import org.junit.Test
+import org.mockito.InjectMocks
+import org.testeditor.dsl.common.testing.AbstractTest
+
+import static com.google.common.io.CharSource.*
+
+class TestLogGroupBuilderTest extends AbstractTest {
+
+	@InjectMocks
+	TestLogGroupBuilder logGroupBuilder
+
+	@Test
+	def void testCreateSystemComponentFirst() {
+		// given
+		val logLine = "[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---"
+
+		// when
+		val group = logGroupBuilder.build(#[logLine])
+
+		// then	
+		assertTrue(group.get(0).type === TestElementType.SystemGroup)
+		assertTrue(group.get(0).logLines.contains(logLine))
+	}
+
+	@Test
+	def void testCreateComponentGroupAfterSystemComponent() {
+		// given
+		val log = '''
+			[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---
+			18:49:10 TRACE [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Component] TestEditorServices
+		'''
+
+		// when
+		val group = logGroupBuilder.build(wrap(log).readLines)
+
+		// then	
+		assertTrue(group.get(1).type === TestElementType.TestComponentGroup)
+	}
+
+	@Test
+	def void testCreateTestSpecGroupAfterSystemComponent() {
+		// given
+		val log = '''
+			[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---
+			18:49:10 INFO  [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Test specification] * Given
+		'''
+
+		// when
+		val group = logGroupBuilder.build(wrap(log).readLines)
+
+		// then	
+		assertTrue(group.get(1).type === TestElementType.TestSpecGroup)
+	}
+
+	@Test
+	def void testCreateComponentGroupAfterTestSpecGroup() {
+		// given
+		val log = '''
+			[INFO] --- xtend-maven-plugin:2.10.0:testCompile (default) @ org.testeditor.rcp4.uatests ---
+			18:49:10 INFO  [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Test specification] * Given
+			18:49:10 TRACE [WorkbenchTestable] [TE-Test: AmlTemplateTest] AbstractTestCase  [Component] TestEditorServices
+		'''
+
+		// when
+		val group = logGroupBuilder.build(wrap(log).readLines)
+
+		// then	
+		assertEquals(group.size, 2)
+		assertTrue(group.get(1) instanceof TestLogGroup)
```

Comment:
maybe use:
val tlGroup = group.get(1).assertInstanceOf(TestLogGroupComposite)  // asserts and returns as TestLogGroupComposite